presto-main/src/main/java/com/facebook/presto/type/BigintOperators.java

```diff
@@ -260,10 +261,10 @@ public static long castToReal(@SqlType(StandardTypes.BIGINT) long value)
     @ScalarOperator(CAST)
     @LiteralParameters("x")
     @SqlType("varchar(x)")
-    public static Slice castToVarchar(@SqlType(StandardTypes.BIGINT) long value)
+    public static Slice castToVarchar(@LiteralParameter("x") long x, @SqlType(StandardTypes.BIGINT) long value)
     {
         // todo optimize me
-        return utf8Slice(String.valueOf(value));
+        return truncateAsciiSliceToLength(utf8Slice(String.valueOf(value)), (int) x);
```

Comment:
Does this fix "length()" function too as the length has been fixed here?

Reply:
Yes, it does.